src/ucp/core/ucp_context.c

```diff
@@ -1184,6 +1187,25 @@ static ucs_status_t ucp_fill_resources(ucp_context_h context,
         }
     }
 
+    /* Check if mem type support is requested w/o its basic detection TLS */
+    for (md_index = 0; md_index < context->num_mds; md_index++) {
```

Comment:
1. i'm pretty sure this check can be simplified by uniting with the construction of context->mem_type_detect_mds inside ucp_add_component_resources(). maybe check mem_type_bitmap instead of a nested for loop?
2. it seems weird that the mere support of "access_mem_type" (which in fact means that gpu-direct driver is loaded, right?) means the user wanted to use it. What if we have a system with gpudirect driver loaded, but no cuda libraries installed? i guess this warning will be printed?

Reply:
comment 2 is irrelevant